install_files/ansible-base/roles/validate/tasks/validate_tails_environment.yml

```diff
@@ -26,3 +26,54 @@
       The SecureDrop git repository should be cloned
       to `~/Persistent/securedrop`.
   with_items: "{{ tails_persistence_check_result.results }}"
+
+- name: Check for v3 client auth files
+  stat:
+    path: "/home/amnesia/Persistent/securedrop/install_files/ansible-base/{{ item }}"
+  register: v3_client_auth_files
+  with_items:
+    - app-ssh.auth_private
+    - app-journalist.auth_private
+    - mon-ssh.auth_private
+
+- name: Count the number of v3 client auth files
+  set_fact:
+    v3_client_auth_file_count: "{{ v3_client_auth_files.results | selectattr('stat.exists') | list | count }}"
+
+- name: Check for Tor v3 key file
+  stat:
+    path: "/home/amnesia/Persistent/securedrop/install_files/ansible-base/tor_v3_keys.json"
+  register: v3_tor_key
+
+- name: Confirm that v3 auth files are present for SSH-over-LAN configuration
+  assert:
+    that:
+      - v3_client_auth_file_count == "0" or v3_client_auth_file_count == "1"
+    msg: >-
+      There are multiple `.auth_private` files under
+      `~/Persistent/securedrop/install_files/ansible-base`, most likely from
+      a previous installation. Please remove any `.auth_private` files except
+      for `app-journalist.auth_private`, and retry the install command.
+  when: v3_onion_services == True and enable_ssh_over_tor == False
```

Comment:
consider this scenario:

1. User has a SecureDrop on 0.14.0 using SSH over Tor (has 0 .auth_private files)
2. User updates to 1.0.0, adds v3 onion service support (now has 3 .auth_private files)
3. User then decides to turn off SSH over Tor and enable SSH over LAN at some later date. The validation here will fail since they have 3 `.private_auth` files in their admin workstation when they run the playbook. 

suggestion: we relax the validation here to allow 0, 1 or 3 .auth_private files

Reply:
I think the appropriate thing to do might be to test for the ssh auth files separately, and if they're running `install` with ssh-over-lan selected, those files should not be present (so prompt them to move them away). `tailsconfig` doesn't reference `site-specific` (because JWs won't have it) so we can't check the  `enable-ssh-over-tor` toggle, and they'll get added to Tor's config if they're present, which we probably don't want.

Reply:
So we'd have the following scenario:

1. User installs their SecureDrop on 0.14.0 using SSH over Tor
2. They add v3 onion service support
3. They run `securedrop-admin tailsconfig`
4. They run `securedrop-admin sdconfig` to turn off SSH over LAN
5. They run `securedrop-admin install` to apply the change (connecting over Tor for the playbook run)
6. The install fails, and instructs them to move their SSH `.auth_private` files (move to somewhere that will persist but not delete them). 
7. They re-run `securedrop-admin install`

I'm concerned if the install fails in step 7 for some other reason (as we know there are all sorts of reasons why this might fail) and the admin runs tailsconfig again at some later date, they might end up in a confusing state unable to SSH in.

Reply:
OK, simplest set of checks that cover all above scenarios:
- confirm that there are 0 or 2 SSH auth files in all cases
- confirm that there is a Journalist auth file if there are 2 SSH auth files and v3 enabled and ssh-over-tor enabled.
- confirm that there is a keys file if there is a journalist auth file and v3 enabled.